lib/spack/spack/filesystem_view.py

```diff
@@ -884,3 +888,10 @@ def get_dependencies(specs):
 
 class ConflictingProjectionsError(SpackError):
     """Raised when a view has a projections file and is given one manually."""
+
+
+def is_folder_on_case_insensitive_filesystem(path: str) -> bool:
+    with tempfile.NamedTemporaryFile(dir=path, prefix=".sentinel") as sentinel:
+        return os.path.exists(
+            os.path.join(path, os.path.join(path, os.path.basename(sentinel.name).upper()))
+        )
```

Comment:
```suggestion
    with tempfile.NamedTemporaryFile(dir=path, prefix=".sentinel"):
        return os.path.exists(os.path.join(path, ".SENTINEL"))
```

remove the double os.path.join and just explicitly writing .SENTINEL is imo more clear than the current basename / upper obfuscation.

Reply:
Fixed.